Assets/Mirror/Runtime/NetworkIdentity.cs

```diff
@@ -33,14 +33,14 @@ public sealed class NetworkIdentity : MonoBehaviour
 
         // properties
         public bool isClient { get; internal set; }
-        public bool isServer => m_IsServer && NetworkServer.active; // dont return true if server stopped.
+        public bool isServer { get => m_IsServer && NetworkServer.active && netId != 0; internal set => m_IsServer = value; } // dont return true if server stopped.
         public bool isLocalPlayer { get; private set; }
         public bool hasAuthority { get; private set; }
 
         // <connectionId, NetworkConnection>
         public Dictionary<int, NetworkConnection> observers;
 
-        public uint netId { get; private set; }
+        public uint netId { get; internal set; }
```

Comment:
please do only netid internal set with the below code and remove setnetworkinstanceid.
we can worry about isserver property afterwards :)

Reply:
No, these two changes go in tandem. Notice how the if (netId == 0) is combined into the isServer. This was Paul's suggestion to my original PR which I closed because I busted it accidentally.

Reply:
Yes vis,  see the stuff below,  the if was folded into a && netId != 0